packages/aws-cdk/lib/docker.ts

```diff
@@ -0,0 +1,208 @@
+import { ContainerImageAssetMetadataEntry } from '@aws-cdk/cx-api';
+import { CloudFormation } from 'aws-sdk';
+import crypto = require('crypto');
+import { ToolkitInfo } from './api/toolkit-info';
+import { debug, print } from './logging';
+import { shell } from './os';
+import { PleaseHold } from './util/please-hold';
+
+/**
+ * Build and upload a Docker image
+ *
+ * Permanently identifying images is a bit of a bust. Newer Docker version use
+ * a digest (sha256:xxxx) as an image identifier, which is pretty good to avoid
+ * spurious rebuilds. However, this digest is calculated over a manifest that
+ * includes metadata that is liable to change. For example, as soon as we
+ * push the Docker image to a repository, the digest changes. This makes the
+ * digest worthless to determe whether we already pushed an image, for example.
+ *
+ * As a workaround, we calculate our own digest over parts of the manifest that
+ * are unlikely to change, and tag based on that.
+ */
+export async function prepareContainerAsset(asset: ContainerImageAssetMetadataEntry, toolkitInfo: ToolkitInfo): Promise<CloudFormation.Parameter[]> {
+  debug(' 👑  Preparing Docker image asset:', asset.path);
+
+  const buildHold = new PleaseHold(` ⌛ Building Docker image for ${asset.path}; this may take a while.`);
+  try {
+    buildHold.start();
+
+    const command = ['docker',
+      'build',
+      '--quiet',
```

Comment:
Do not pass `--quiet` if `--verbose` is used

Reply:
I am not 100% sure we don't want to show this output. It can take a looong time

Reply:
`--quiet` is the way to make Docker output machine-friendly output (image ID of the just-produced image).

Without quiet, it prints human-centric garble that contains strings that we need to parse the image ID out of, and no idea how stable that is.

(If you're going to look: there are some options for this in the Docker manual but those only seem to apply to quite recent versions. At least they don't work on mine. I was aiming for compatibility here.)

My original implementation did have the implementation you proposed, but it makes it harder.

Reply:
does it print it to STDOUT or STDERR?